### PR TITLE
View pr

### PR DESCRIPTION
**Pull request by daniellandau**
Related to https://github.com/clearlydefined/website/pull/219

### CODE REVIEW

File: routes/curations.js [hunk 1]
```diff
@@ -6,6 +6,12 @@ const express = require('express')
 const router = express.Router()
 const utils = require('../lib/utils')
 
+router.get('/base_url', (request, response) => {
+  const repo = request.app.locals.config.curation.store.github.repo
```
Comment (jeffmcaffer) [inline routes/curations.js]: suggest this be removed in favor of returning this value in the `/pr` route response. See other comments

File: routes/curations.js [hunk 2]
```diff
@@ -18,6 +24,19 @@ router.get(
   })
 )
 
+// Get list of components included in a pr
+router.get(
+  '/pr/:pr',
+  asyncMiddleware(async (request, response) => {
+    return curationService.getPrComponents(request.params.pr).then(result => {
```
Comment (jeffmcaffer) [inline routes/curations.js]: Here add in the URL for the PR so we can avoid people constructing URLs. So the response might look like 

```javascript
{
  pr: 107,
  url: 'https://github.com/....',
  changes: [
    npm/npmjs/-/redie/0.3.0,
    git/github/microsoft/redie/082303205723abc324bd3e
  ]
}
```

That makes the response a bit more stand alone/complete

File: providers/curation/github.js [hunk 3]
```diff
@@ -320,6 +328,25 @@ class GitHubCurationService {
     }
   }
 
+  async getPrComponents(number) {
```
Comment (jeffmcaffer) [inline providers/curation/github.js]: Nice. Some naming comments (naming is hard)

* getPrComponents => getChangedDefinitions
* result => files
* prComponents => changedCoordinates
* fileObj => remove and just files[i] in the one place
* prVersion => prDefinitions
* masterVersion => currentDefinitions
* relevantRevisions => changedRevisions

File: providers/curation/github.js [hunk 4]
```diff
@@ -93,14 +93,23 @@ class GitHubCurationService {
         )
       )
     )
-    return (userGithub || serviceGithub).pullRequests.create({
+    const retVal = await (userGithub || serviceGithub).pullRequests.create({
```
Comment (jeffmcaffer) [inline providers/curation/github.js]: general naming comment. we tend to use whole words and semantic naming. So `retVal` => `result` and `commentObj` => `comment`

File: routes/curations.js (hunk 2 above)
Comment (daniellandau) [reply routes/curations.js]: I think that depends on whether or not the base URL is useful elsewhere as well. In the currently disabled `PageCurate` that same URL is hard coded and could be replaced with an API call, but the `/curate` page only shows one component so it's not exactly the same and thus doesn't need the same full dataset either.

File: providers/curation/github.js (hunk 4 above)
Comment (daniellandau) [reply providers/curation/github.js]: Naming updated.

File: routes/curations.js (hunk 1 above)
Comment (jeffmcaffer) [reply routes/curations.js]: @daniellandau thoughts on this? I commented similarly in the corresponding website PR.

File: providers/curation/github.js [hunk 5]
```diff
@@ -93,14 +93,23 @@ class GitHubCurationService {
...
       body: patch.description,
       head: `refs/heads/${prBranch}`,
       base: branch
     })
+    const number = result.data.number
+    const comment = {
+      owner,
+      repo,
+      number,
+      body: `You can review the change introduced to the full definiton at [ClearlyDefined](https://clearlydefined.io/view_pr/${number}).`
```
Comment (jeffmcaffer) [inline providers/curation/github.js]: this url changed. It is now 'contributions'

also, type in 'definition' (missing the last i)

Comment (jeffmcaffer) [reply providers/curation/github.js]: Note that I merged this PR and fixed the above issue

**Pull request merged**